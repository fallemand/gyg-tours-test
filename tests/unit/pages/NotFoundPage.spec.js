import React from 'react';
import NotFoundPage from '../../../app/pages/not-found/NotFoundPage';

describe('NotFoundPage', () => {
  const props = {
    history: { push: jest.fn() },
    match: { params: { id: 1 } },
  };

  it('render component', () => {
    const component = shallow(<NotFoundPage {...props} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('click on button should redirect to tours', () => {
    const component = shallow(<NotFoundPage {...props} />);
    component.find('button').simulate('click');
    expect(props.history.push).toHaveBeenCalledWith('/tours');
  });
});
