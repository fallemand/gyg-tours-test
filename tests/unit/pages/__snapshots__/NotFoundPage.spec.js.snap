// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage render component 1`] = `"<ErrorPage title='404' message='We couldn\\\\'t find the page you\\\\'re looking for :(' buttonLabel='Go Home!' onClick={[Function: onClick]} />"`;
